templates/web_gateway/web_gateway.py

```diff
@@ -116,28 +106,41 @@ def _remove_zt_clients(self, nics, node_url):
             data = {'url': node_url, 'name': node_zt_name}
             zt_client.schedule_action('remove_from_robot', args=data).wait(die=True)
 
+    def _install_etcd_cluster(self):
+        self.logger.info('Installing etcd cluster')
+        cluster_data = {
+            'nrEtcds': self.data['nrEtcds'],
+            'password': self.data['etcdPassword'],
+            'farmerIyoOrg': self.data['farmerIyoOrg'],
+            'nics': self.data['nics'],
+        }
+        etcd_cluster = self.api.services.find_or_create(ETCD_CLUSTER_TEMPLATE_UID, self._etcds_name, cluster_data)
+        etcd_cluster.schedule_action('install').wait(die=True)
+        cluster_connection = etcd_cluster.schedule_action('connection_info').wait(die=True).result
+        return cluster_connection
+
     def _install_traefik(self, traefik_endpoint):
         self.logger.info('Installing traefik')
-        nics = self._create_zt_clients(self.data['nics'], self._traefik_url)
+        nics = self._create_zt_clients(self.data['nics'], self._public_url)
         data = {
             'etcdEndpoint': traefik_endpoint,
             'etcdPassword': self.data['etcdPassword'],
-            'etcdWatch': self.data['etcdWatch'],
+            'etcdWatch': True,
```

Comment:
You dont have to send this, it already defaults to True.

Reply:
better to make it explicit so we're covered if the default value changes